rosetta-source/src/main/rosetta/legaldocumentation-csa-type.rosetta

```diff
@@ -971,50 +996,98 @@ type CoveredTransactions: <"Specification of Transactions covered by the legal a
+    threshold Threshold (1..1) <"Represents the amount of unsecured risk that a party is willing to tolerate without holding any posted collateral provided by the other party.">
         [docReference ISDA CSA_NewYork_1994]
         [docReference ISDA CSA_English_1995]
         [docReference ISDA CSD_English_1995]
         [docReference ISDA CSA_VM_English_2016]
         [docReference ISDA CSA_VM_NewYork_2016]
-    minimumTransferAmount MinimumTransferAmount (0..1) <"Represents the threshold specified in the agreement below which collateral need not be delivered, and represents the smallest incremental value of exposure that must be reached before the parties are obliged to transfer collateral.">
+    rounding CollateralRounding (1..1) <"The rounding methodology applicable to the Delivery Amount and the Return Amount in terms of nearest integral multiple of Base Currency units.">
+    independentAmount LegacyIndependentAmount (1..1) <"Specifies the additional amount of collateral that is required to be delivered by a party irrespective of the mark-to-market exposure under the related Transactions. It may take the form of a fixed amount or may be determined using ratings.">
+        [docReference ISDA CSA_NewYork_1994]
+        [docReference ISDA CSA_English_1995]
+        [docReference ISDA CSD_English_1995]
+        [docReference ISDA CSA_VM_English_2016]
+        [docReference ISDA CSA_VM_NewYork_2016]
+        [docReference ISDA CSD_IM_English_2016]
+        [docReference ISDA CSA_IM_NewYork_2016]
+    creditSupportAmount CreditSupportAmount (1..1) <"The total amount one counterparty must deliver to the other at any time: the combination of the Exposure to that party and the net Independent Amounts it must post, minus any agreed Threshold.">    
+    eligibleCreditSupport EligibleCreditSupport (1..1) <"Specification of the Eligible Collateral or Credit Support applicable to the Agreement.">
+        [docReference ISDA CSA_NewYork_1994]
+        [docReference ISDA CSA_English_1995]
+        [docReference ISDA CSD_English_1995]
+        [docReference ISDA CSA_VM_English_2016]
+        [docReference ISDA CSA_VM_NewYork_2016]    
+
+type CreditSupportObligationsIM extends CreditSupportObligationsBase: <"Specification of the Credit Support Obligations applicable to the Initial Margin Credit Support Annex.">
+    [docReference ISDA CSD_IM_English_2016 paragraph "13 General Principles" clause "(c)" name "Credit Support Obligations"]
+    [docReference ISDA CSA_IM_Japanese_2016 paragraph "13 General Principles" clause "(d)" name "Credit Support Obligations"]
+    [docReference ISDA CSA_IM_NewYork_2016 paragraph "13 General Principles" clause "(c) " name "Credit Support Obligations"]
+    marginApproach MarginApproach (1..1) <"The selection of Margin Approach applicable to the agreement.">
+    otherEligibleSupportIM string (0..1) <"The Other Eligible Support election. If not specified, this election is deemed as not applicable.">
```

Comment:
Does this need IM on it now? Due to the fact it is just in the Initial Margin Credit Support Obligations, could it not be just `otherEligibleSupport`?